plugins/postgres/dbt/include/postgres/macros/materializations/materialized_view.sql

```diff
@@ -0,0 +1,30 @@
+{% macro postgres__get_create_materialized_view_as_sql(relation, sql) %}
+    {{ return(get_create_view_as_sql(relation, sql)) }}
+{% endmacro %}
+
+
+{% macro postgres__get_refresh_data_in_materialized_view_sql(relation) %}
+    select 1;
+{% endmacro %}
+
+
+{% macro postgres__get_replace_materialized_view_as_sql(relation, sql, existing_relation, backup_relation, intermediate_relation) %}
+    {{ get_create_view_as_sql(intermediate_relation, sql) }}
+
+    {% if existing_relation is not none %}
+        alter view {{ existing_relation }} rename to {{ backup_relation.include(database=False, schema=False) }};
+    {% endif %}
+
+    alter view {{ intermediate_relation }} rename to {{ relation.include(database=False, schema=False) }};
+
+{% endmacro %}
+
+
+{% macro postgres__get_alter_materialized_view_sql(relation, updates, sql, existing_relation, backup_relation, intermediate_relation) %}
+    {% if 'index' in updates.keys() %}
+    --  should this be placed lower into the chain somehow to be below call to main? othe
+        {% do create_indexes(target_relation) %}
```

Comment:
I am a little concerned about this placement most other materializations that call do it a little below the call to `main` for statement builds but not sure if we want this just in the base materialization or to recreate the materialization for this small addition in postgres

Reply:
Oh, good point, in other materializations we call the macro to create the indexes after the object is created. However, that's the change that we're looking for in this case, so we also call it during the build. I think we would need to put that in the create and replace steps then, in the postgres versions. Let's leave that for the native implementation story since creating indexes on the proxy view doesn't make sense.